File: backend/src/main/java/com/rythulink/backend/service/OrderService.java
package com.rythulink.backend.service;

import com.rythulink.backend.model.Order;

import java.util.List;

public interface OrderService {

    Order placeOrder(Order order);

    List<Order> getAllOrders();

    List<Order> getBuyerOrders(String buyerId);

    List<Order> getFarmerOrders(String farmerId);

    Order updateOrderStatus(String orderId, String status);
}

File: frontend/src/services/orderService.js
import api from "./api";

export const placeOrder = async (orderData) => {
  return await api.post("/api/orders", orderData);
};

export const getBuyerOrders = async (buyerId) => {
  return await api.get(`/api/orders/buyer/${buyerId}`);
};

export const getFarmerOrders = async (farmerId) => {
  return await api.get(`/api/orders/farmer/${farmerId}`);
};

export const updateOrderStatus = async (orderId, status) => {
  return await api.put(`/api/orders/${orderId}/status/${status}`);
};